activesupport/lib/active_support/core_ext/object/acts_like.rb

```diff
@@ -1,11 +1,32 @@
 # frozen_string_literal: true
 
 class Object
-  # A duck-type assistant method. For example, Active Support extends Date
-  # to define an <tt>acts_like_date?</tt> method, and extends Time to define
-  # <tt>acts_like_time?</tt>. As a result, we can do <tt>x.acts_like?(:time)</tt> and
-  # <tt>x.acts_like?(:date)</tt> to do duck-type-safe comparisons, since classes that
-  # we want to act like Time simply need to define an <tt>acts_like_time?</tt> method.
+  # Provides a way to check whether some class acts like some other class based on the existence of
+  # an appropriately-named marker method.
+  #
+  # A class that provides the same interface as <tt>X</tt> may define a marker method named
+  # <tt>acts_like_X?</tt> to signal its compatibility.
+  #
+  # For example, Active Support extends Date to define an <tt>acts_like_date?</tt> method, and
+  # extends Time to define <tt>acts_like_time?</tt>. As a result, we can do
+  # <tt>x.acts_like?(:time)</tt> and <tt>x.acts_like?(:date)</tt> to do duck-type-safe comparisons,
+  # since classes that we want to act like Time simply need to define an <tt>acts_like_time?</tt>
+  # method
+  #
+  # Note that the marker method is only expected to exist. It isn't called, so its body or return
+  # value are irrelevant.
+  #
+  # ==== Example: A class that provides the same interface as <tt>String</tt>
+  #
+  # This class may define:
+  #
+  #   def acts_like_string?
+  #   end
+  #
+  # Then client code can query for duck-type-safeness this way:
+  #
+  #   some_klass.acts_like?(:string)
```

Comment:
What about making this example more concrete?

```suggestion
  #   class Stringish
  #     def acts_like_string?
  #     end
  #   end
  #
  #   Stringish.new.acts_like?(:string) # => true
```

Reply:
Great suggestion, I've updated the PR.